components/net/filemanager_thread.rs

```diff
@@ -414,6 +434,93 @@ impl FileManagerStore {
         self.get_blob_buf(sender, &id, &origin_in, RelativePos::full_range(), check_url_validity)
     }
 
+    fn fetch_blob_buf(&self, sender: &servo_channel::Sender<Data>,
+                    id: &Uuid, origin_in: &FileOrigin, rel_pos: RelativePos,
+                    check_url_validity: bool, response: &mut Response) -> Result<(), BlobURLStoreError> {
+        let mut bytes = vec![];
+        let file_impl = self.get_impl(id, origin_in, check_url_validity)?;
+        match file_impl {
+            FileImpl::Memory(buf) => {
+                let range = rel_pos.to_abs_range(buf.size as usize);
+                let blob_buf = BlobBuf {
+                    filename: None,
+                    type_string: buf.type_string,
+                    size: range.len() as u64,
+                    bytes: buf.bytes.index(range).to_vec(),
+                };
+
+                let content_type: Mime = blob_buf.type_string.parse().unwrap_or(mime!(Text / Plain));
+                let charset = content_type.get_param(Attr::Charset);
+                let mut headers = Headers::new();
+
+                if let Some(name) = blob_buf.filename {
+                    let charset = charset.and_then(|c| c.as_str().parse().ok());
+                    headers.set(ContentDisposition {
+                        disposition: DispositionType::Inline,
+                        parameters: vec![
+                            DispositionParam::Filename(charset.unwrap_or(Charset::Us_Ascii),
+                                                       None, name.as_bytes().to_vec())
+                        ]
+                    });
+                }
+
+                headers.set(ContentLength(blob_buf.size as u64));
+                headers.set(ContentType(content_type.clone()));
+
+                bytes.extend_from_slice(&blob_buf.bytes);
+
+                response.headers = headers;
+                *response.body.lock().unwrap() = ResponseBody::Done(bytes);
```

Comment:
This needs to send bytes over `sender` instead of modifying the response object's body.